### PR TITLE
Ft refactor friendly invoice

### PR DESCRIPTION
**Pull request by Awath-Abdat**
# Description

Changed the invoice ID into a more user friendly 12 digit ID

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Trello Ticket ID

https://trello.com/c/x9brklA7

## How Can This Be Tested?

Run python manage.py db upgrade to update the database for the invoice ID was in UUID format and had to change it to a string then send yourself an invoice email and you will have the glorious user friendly ID in the email.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
Latest Invoice email

### CODE REVIEW

File: app/controllers/project.py [hunk 1]
```diff
@@ -128,7 +130,7 @@ def post(self):
                         message='Internal Server Error'), 500
 
             # create a billing invoice on project creation
-            new_invoice = BillingInvoice(project_id=project.id)
+            new_invoice = BillingInvoice(id="CC"+str(time.strftime("%y", time.localtime()))+"-"+str(uuid.uuid4())[:8], project_id=project.id)
```
Comment (Mubangizi) [inline app/controllers/project.py:133]: Hey @Awath-Abdat, do you mind elaborating on what this line does., why the changes?

File: app/controllers/transactions.py [hunk 2]
```diff
@@ -54,7 +56,7 @@ def post(self, project_id):
             tx_ref = validated_transaction_data['tx_ref']
             # comments for implementation flow
             # get the latest invoice for a project by date
-            invoice = BillingInvoice.query.filter_by(project_id=project_id, is_cashed=False).order_by(
+            invoice = BillingInvoice.query.filter_by(id="CC"+str(time.strftime("%y", time.localtime()))+"-"+str(uuid.uuid4())[:8], project_id=project_id, is_cashed=False).order_by(
```
Comment (Mubangizi) [inline app/controllers/transactions.py:59]: And this too

File: app/models/billing_invoice.py [hunk 3]
```diff
@@ -14,7 +14,7 @@ class BillingInvoice(ModelMixin):
     __tablename__ = 'billing_invoices'
 
     # billing invoice fields
-    id = db.Column(UUID(as_uuid=True), primary_key=True, 
+    id = db.Column(db.String, primary_key=True,
```
Comment (Mubangizi) [inline app/models/billing_invoice.py:17]: Please Change this back to UUID

File: app/models/billing_metrics.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class BillingMetrics(ModelMixin):
                    server_default=sa_text("uuid_generate_v4()"))
     app_id = db.Column(UUID(as_uuid=True),
                        db.ForeignKey('app.id'), nullable=False)
-    invoice_id = db.Column(UUID(as_uuid=True), db.ForeignKey(
+    invoice_id = db.Column(db.String, db.ForeignKey(
```
Comment (Mubangizi) [inline app/models/billing_metrics.py:18]: Change this back to UUID

File: app/models/transaction_record.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ class TransactionRecord(ModelMixin):
     id = db.Column(UUID(as_uuid=True), primary_key=True, server_default=sa_text("uuid_generate_v4()"))
     owner_id = db.Column(UUID(as_uuid=True), db.ForeignKey('user.id'), nullable=False)
     project_id = db.Column(UUID(as_uuid=True), db.ForeignKey('project.id'), nullable=False)
-    billing_invoice_id = db.Column(UUID(as_uuid=True), db.ForeignKey('billing_invoices.id'), nullable=True)
+    billing_invoice_id = db.Column(db.String, db.ForeignKey('billing_invoices.id'), nullable=True)
```
Comment (Mubangizi) [inline app/models/transaction_record.py:18]: Change this back to UUID

**Review by Mubangizi — CHANGES_REQUESTED** [review]
Hey @Awath-Abdat, Great work on this. For consistency purposes,, we use UUIDs for all the modals we have in the database, so please revert the IDs to UUID

File: app/controllers/project.py (hunk 1 above)
Comment (Awath-Abdat) [reply app/controllers/project.py:133]: A custom ID was needed so every time a new invoice is created, i made the custom ID to be sent along during the creation of the billing invoice record. I didn't find a way of making the custom ID of the format CCYY-1a2b3c4d to be created by default on postgres db when the record is being created.

Comment (Mubangizi) [reply app/controllers/project.py:133]: OOh makes sense, alright, so create a different field o=in the invoice table for the custom id in the DB. An that's for display

Comment (e-ian) [reply app/controllers/project.py:133]: @Awath-Abdat could you spare a moment and work on this feedback by creating a different field in the invoice table to display the formatted invoice id.

File: app/controllers/billing_invoice.py [hunk 6]
```diff
@@ -82,6 +84,7 @@ def post(self, project_id):
 
 
             new_invoice_data = dict(
+                id="CC"+str(time.strftime("%y", time.localtime()))+"-"+str(uuid.uuid4())[:8],
```
Comment (e-ian) [inline app/controllers/billing_invoice.py:87]: nice!

**Review by e-ian — CHANGES_REQUESTED** [review]

File: app/controllers/project.py (hunk 1 above)
Comment (Awath-Abdat) [reply app/controllers/project.py:133]: Am going to make the change, so i should create a seperate field for this new invoice ID instead of making changes to the original UUID ID which we were using

Comment (e-ian) [reply app/controllers/project.py:133]: You can revert the invoice_id field back to uuid  and create a new fields called invoice_display_id that will be displayed in the emailed invoice. We shall maintain the invoice_id in UUID form. Make sense? @Awath-Abdat

Comment (e-ian) [reply app/controllers/project.py:133]: Is it possible to push these changes before the demo @Awath-Abdat .

Comment (Awath-Abdat) [reply app/controllers/project.py]: I have adjusted and made the changes, i was in a paper today morning so worked on it after the paper

**Review by e-ian — APPROVED** [review]
